src/ray/object_manager/plasma/client.cc

```diff
@@ -445,8 +447,15 @@ Status PlasmaClient::Impl::ExperimentalMutableObjectWriteAcquire(
                                    ") is larger than allocated buffer size " +
                                    std::to_string(entry->object.allocated_size));
   }
-  plasma_header->WriteAcquire(
-      entry->next_version_to_write, data_size, metadata_size, num_readers);
+  // TODO(ekl) should we just only track the version in the plasma header?
```

Comment:
@stephanie-wang is there any reason we shouldn't switch to incrementing the plasma header value? It seems less error prone than duplicating the version counter.

Reply:
Yeah in the original draft, we didn't have a semaphore to protect against multiple writers, so it wasn't safe. But I think with the semaphore, it should work now to just wait on the semaphore, and then increment whatever version is there.

Reply:
Would that simplify this PR at all? I think it is a simple enough fix that we could include it here.

Reply:
+1 on this

Reply:
Yeah, I could roll this in. So WriteAcquire would internally figure out the next version id, correct?

Reply:
Yes that sounds good.